libs/s25main/desktops/dskCredits.cpp

```diff
@@ -335,6 +479,7 @@ void dskCredits::SetActive(bool active)
     Desktop::SetActive(active);
     if(active)
     {
-        startTime = bobTime = bobSpawnTime = VIDEODRIVER.GetTickCount();
+        bobTime = bobSpawnTime = VIDEODRIVER.GetTickCount();
```

Comment:
This needs to change too as it is the current source of the bug which @Spikeone mistakenly thought of a "broken skip": There was no skipping it (re)activated the desktop on clicking it which restarts the timers.

It should only happen on the first call, or (now easier): When the timer isn't already running (as we never stop it it works)

Reply:
doesnt happen for me now anymore?

Reply:
It still should as it is still wrong: The SetActive may happen multiple times caused by clicking. The `timer.restart` below will cause the `time` in `DrawCredits` to reset and the bobTime/bobSpawnTime reset above will cause an additional delay in spawning bobs and a lack in their animation (which likely won't be noticed though)
I'd suggest to use a timer for all 3, and here check if the timer has been started and only if not call start

Reply:
the only thing which will happen is that you stop spawning new bobs if you click. Thats even kind of an easter egg ;-)
![Credits](https://user-images.githubusercontent.com/656249/102721738-04d29200-42fd-11eb-8198-d625554101bf.gif)

Reply:
It also has this skip-like behavior @Spikeone mentioned which is rather odd. However it looks like it is actually skipping now, is it? I'm surprised by that as from the code I think it just restarts the current credit. If it does skip, it is fine

Reply:
it steps to the next credit on click now (and resets the time a new bob will be spawned, so its working